src/materials/ComputeIsotropicElasticityTensorSoil.C

```diff
@@ -20,32 +20,62 @@ validParams<ComputeIsotropicElasticityTensorSoil>()
   InputParameters params = validParams<ComputeElasticityTensorBase>();
   params += validParams<LayeredMaterialInterface<>>();
   params.addClassDescription("Compute an isotropic elasticity tensor for a "
-                             "layered soil material when shear modulus, "
+                             "layered soil material when shear modulus or elastic modulus, "
                              "poisson's ratio and density are provided as "
                              "input for each layer.");
   params.setDocString("layer_ids",
                       "Vector of layer ids that map one-to-one "
-                      "with the 'shear_modulus', 'poissons_ratio' "
+                      "with the 'shear_modulus' or 'elastic_modulus', 'poissons_ratio' "
                       "and 'density' input parameters.");
   params.addRequiredParam<std::vector<Real>>("poissons_ratio",
                                              "Vector of Poisson's ratio values that map one-to-one "
                                              "with the number 'layer_ids' parameter.");
-  params.addRequiredParam<std::vector<Real>>("shear_modulus",
-                                             "Vector of  shear modulus values that map one-to-one "
-                                             "with the number 'layer_ids' parameter.");
+  params.addParam<std::vector<Real>>("shear_modulus",
+                                     "Vector of shear modulus values that map one-to-one "
+                                     "with the number 'layer_ids' parameter.");
+  params.addParam<std::vector<Real>>("elastic_modulus",
+                                     "Vector of elastic modulus values that map one-to-one "
+                                     "with the number 'layer_ids' parameter.");
   params.addRequiredParam<std::vector<Real>>(
       "density",
       "Vector of density values that map one-to-one with the number "
       "'layer_ids' parameter.");
   params.addParam<bool>(
-      "wave_speed_calculation", true, "Set to False to turn off P and S wave speed calcualtion.");
+      "wave_speed_calculation", true, "Set to False to turn off P and S wave speed calculation.");
   return params;
 }
 
+const MooseArray<Real> &
+getShearModulus(ComputeIsotropicElasticityTensorSoil * object,
+                std::vector<Real> & shear_modulus,
+                std::string name)
```

Comment:
One more const I just noticed, this can be `const std::string & name`

Reply:
I will make this change as well while I am at it.